src/Http/Http.Extensions/test/ProblemDetailsDefaultWriterTest.cs

```diff
@@ -600,6 +600,8 @@ await writer.WriteAsync(new ProblemDetailsContext()
     [InlineData("application/*")]
```

Comment:
Are we meant to be able to write this at all then? cc @captainsafia

Reply:
According to the RFC, it looks like `applicaton/*`is a content-type that we should match against. 

> The asterisk "*" character is used to group media types into ranges, with "*/*" indicating all media types and "type/*" indicating all subtypes of that type. The media-range can include media type parameters that are applicable to that range.

I believe some of the weirdness here might be related to the way that comparisons happen for parameters in the [IsSubsetOf](https://source.dot.net/#Microsoft.Net.Http.Headers/MediaTypeHeaderValue.cs,385) extension method. Previously, it was checking to see if all the parameters defined in `acceptsHeaderValue` appeared in `_jsonMediaType`. Since `_jsonMediaType` has no parameters defined, this would always return false, hence the inversion eeded.

We need to retain the `_jsonMediaType.IsSubsetOf(acceptHeaderValue)` because `MediaTypeHeaderValue` handles comparisons for subtypes differently (it requires the argument provided to be more permissive).

TL;DR: I think the issue here is the confusing `IsSubsetOf` APIs on `MediaTypeHeaderValue`.